conjur/controller/init_controller.py

```diff
@@ -37,53 +39,61 @@ class InitController:
     init_logic = None
 
     def __init__(self, conjurrc_data: ConjurrcData, init_logic: InitLogic, force: bool,
-                 ssl_verify: bool):
-        self.ssl_verify = ssl_verify
-        if self.ssl_verify is False:
+                 ssl_verification_data: SslVerificationMetadata):
+        self.ssl_verification_data = ssl_verification_data
+
+        if self.ssl_verification_data.is_insecure_mode:
             util_functions.get_insecure_warning_in_debug()
             util_functions.get_insecure_warning_in_warning()
 
+        if self.ssl_verification_data.is_self_signed_mode:
+            self._prompt_warning_for_self_signed_flow()
+
         self.conjurrc_data = conjurrc_data
         self.init_logic = init_logic
         self.force_overwrite = force
 
     def load(self):
         """
         Method that facilitates all method calls in this class
+        In more details this function job is to create the conjurrcData object and write it to disk
         """
         if self.conjurrc_data.conjur_url is None:
-            self._prompt_for_conjur_url()
-
+            self.conjurrc_data.conjur_url = self._prompt_for_conjur_url()
         formatted_conjur_url = self._format_conjur_url()
-        self._validate_conjur_url(formatted_conjur_url, self.ssl_verify)
 
-        if self.ssl_verify is True:
-            fetched_certificate = self._get_server_certificate(formatted_conjur_url)
-            # For a uniform experience, regardless if the certificate is self-signed
-            # or CA-signed, we will write the certificate on the machine
-            self._write_certificate(fetched_certificate)
-        else:
-            self.conjurrc_data.cert_file = ""
+        self._validate_conjur_url(formatted_conjur_url)
 
-        self._get_account_info()
+        self._fetch_certificate_if_needed_and_update_conjurrc(formatted_conjur_url)
+        self._get_account_info_if_not_exist()
         self.write_conjurrc()
-
         sys.stdout.write("Successfully initialized the Conjur CLI\n")
 
+    def _fetch_certificate_if_needed_and_update_conjurrc(self, formatted_conjur_url):
+        mode = self.ssl_verification_data.mode
+        if mode == SslVerificationMode.WITH_CA_BUNDLE:
+            self.conjurrc_data.cert_file = self.ssl_verification_data.ca_cert_path
+        if mode == SslVerificationMode.SELF_SIGN:
+            fetched_certificate = self._get_server_certificate(formatted_conjur_url)
+            self._write_certificate(fetched_certificate)
+        if mode in [SslVerificationMode.NO_SSL, SslVerificationMode.WITH_TRUST_STORE]:
```

Comment:
Is `else` enough?

Reply:
changed to elif incase the enum value is None